.github/workflows/release-e2e.yaml

```diff
@@ -0,0 +1,51 @@
+name: release-e2e
+
+on:
+  release:
+    types: [created]
+  workflow_dispatch:
+
+jobs:
+  release-e2e-kind:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-go@v2
+      with:
+        go-version: '~1.18'
+
+    - name: install_kind_binary
+      env:
+        KIND_VERSION: v0.12.0
+        KIND_OS: kind-linux-amd64
+      run: |
+        curl -sLo kind "https://github.com/kubernetes-sigs/kind/releases/download/${KIND_VERSION}/${KIND_OS}"
+        chmod +x kind
+        sudo mv kind /bin/
+
+    - name: start_kind_cluster
+      run: |
+        # /bin/kind create cluster
+        make kind-cluster
+
+
+    - name: install_kubectl
+      run: |
+        curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+        sudo install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
+
+    - name: release-install
+      needs: [start_kind_cluster, install_kubectl]
+      run: |
+        kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.7.1/cert-manager.yaml
+        kubectl wait --for=condition=Available --namespace=cert-manager deployment/cert-manager-webhook --timeout=60s
+        kubectl apply -f https://github.com/operator-framework/rukpak/releases/download/v0.3.0/rukpak.yaml
```

Comment:
Can we infer the version of the release that we are testing based off the metadata provided in the release? We need to ensure that this step is applying the $latest release at each run.

Reply:
Yup +1 here: we _might_ be able to get away with using the latest endpoint, but we also might be able to infer the tag here as we're only running this workflow during tag event triggers? I think my preference is highly skewed towards the latter implementation. 

The [release workflow](https://github.com/operator-framework/rukpak/blob/main/.github/workflows/release.yaml#L39) has a conditional that checks whether the git reference we're processing is a tag, and parses that value to pull out the tag value. We'd have to update this logic to use a variable in the URL:

```yaml
kubectl apply -f https://github.com/operator-framework/rukpak/releases/download/${IMAGE_TAG}/rukpak.yaml
```

Reply:
@nsapse I think this is the only blocking comment on my end now as we can't hardcode the rukpak version (e.g. v0.3.0 in this case) due to the increase in maintenance over time.

Reply:
I couldn't get the example you linked working so I piped the tags through sort and took the most recent with head. It works but let me know if it's too much of a hack.

Reply:
I _think_ I just had to do something similar to this yesterday: https://github.com/operator-framework/rukpak/blob/main/.github/workflows/release.yaml#L58. You should be able to just parse the tag reference using that `${GITHUB_REF#refs/tags/}` value?

Reply:
For some reason when I do that the URL I get is `https://github.com/operator-framework/rukpak/releases/download/refs/heads/release_actions/rukpak.yaml` i.e. the results of `${GITHUB_REF#refs/tags/}` is `relase_actions`

Reply:
Yeah you'd need to add a conditional that checks whether the current HEAD is a tag reference. You can use the `if` field for conditional statements (and the link I had sent has the conditional you need on line 55).

Reply:
That's with the check pulled from your example. I must be implementing it incorrectly. I'll play around with it.

Reply:
Alright, working.